chain-crypto/src/asymlock.rs

```diff
@@ -20,10 +20,10 @@ pub enum DecryptionError {
 
 fn shared_key_to_symmetric_key(app_level_info: &[u8], p: &GroupElement) -> ChaCha20Poly1305 {
     // use the compressed point as PRK directly
-    #[cfg(feature = "ristretto255")]
+    #[cfg(not(feature = "p256k1"))]
```

Comment:
As I wrote in another PR, I don't like features that implicitly negate alternative dependencies. This is also not scalable: if we add a third backend in the future, we'd have to expand this tedious exclusive logic everywhere.

Can we have both `p256k1` and `ristretto255`, with one enabled by default? It would be OK to have the compilation break if both are enabled, so the test configuration would use `--no-default-features --features p256k1`.

Reply:
Dumb question, is it possible to use `--no-default-features` while targeting only one package in a workspace like I can with `--features chain-crypto/p256k1`?

Reply:
I'm having a hard time making this work in a workspace. Maybe using https://crates.io/crates/cfg-if would make selecting the backend at compile time better?